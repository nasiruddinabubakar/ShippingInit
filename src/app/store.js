import { configureStore } from "@reduxjs/toolkit";
import orderReducer from "../features/orders/orderSlice";
const store = configureStore({

    reducer:orderReducer
});


export default store;