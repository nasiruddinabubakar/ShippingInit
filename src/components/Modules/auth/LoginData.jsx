import { useState } from "react";
import styles from "./SignUp.module.css";
import { InputWithLabel } from "../../Molecules/InputWithLabel";
import {  useNavigate } from "react-router-dom";

import TopDown from "../../framer/TopDown";
import { toast, ToastContainer } from "react-toastify";
import SpinnerFullPage from "../../UI/SpinnerFullPage";
import { postData } from "../../../utils/postData";
import { useDispatch } from "react-redux";
import { addUser } from "../../../features/user/userSlice";


export function LoginData() {
  const [mail, setMail] = useState("");
  const [password, setPassword] = useState("");
  const [isloading, setIsLoading] = useState(false);
  const navigate = useNavigate();
  const dispatch = useDispatch();

  function onhandleEmail(e) {
    setMail(e.target.value);
  }

  function onhandlePassword(e) {
    console.log(e);
    setPassword(e.target.value);
  }
  const validateEmail = (email) => {
    // Regular expression for email validation
    const regex = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/;

    return regex.test(email);
  };
  const notify = (text) => {
    toast.error(text, {
      position: toast.POSITION.TOP_RIGHT,
      className: "toast_message",
    });
  };

  async function handleLogin(e) {
    e.preventDefault();
    if (validateEmail(mail) === false) {
      notify("Email is not valid");
      return;
    } else {
    }
    if (password.length < 4) {
      notify("Enter Valid Password");
      return;
    }
//https://ship-backend-qmsc.onrender.com/api/users/login
    try {
      setIsLoading((isloading) => true);
      const response = await postData("http://127.0.0.1:5000/api/users/login", {
        mail: mail,
        password: password,
      });
      setIsLoading((isloading) => false);
      console.log(response);
      dispatch(addUser(response.user?.user_id));
      if (response.status === "success") {
        localStorage.setItem("user", response.auth);
        moveToNext();
      } else {
        throw new Error("Invalid Email Or Password");
      }
      // Handle the response here if needed.
    } catch (error) {
      if (error) {
        toast.error(error.message, {
          position: toast.POSITION.TOP_RIGHT,
          className: "toast_message",
        });
      }
    } finally {
      setTimeout(() => {
        setIsLoading((isloading) => false);
      }, 200);
    }

    function moveToNext() {
      setIsLoading((isloading) => true);
      setTimeout(() => {
        setIsLoading((isloading) => false);
        navigate("/");
      }, 100);
    }
  }

  return isloading ? (
    <SpinnerFullPage />
  ) : (
    <LoginForm
      onhandleEmail={onhandleEmail}
      onhandlePassword={onhandlePassword}
      handleLogin={handleLogin}
    />
  );
}

export const LoginForm = ({ onhandleEmail, onhandlePassword, handleLogin }) => {
  return (
    <TopDown>
      <form className={styles.login} onSubmit={(e) => handleLogin(e)}>
        <div className={styles.emailinput}>
          <InputWithLabel
            text={"Email Address"}
            type={"email"}
            placeholder={"jack99@gmail.com"}
            onChangeInput={onhandleEmail}
          />
        </div>
        <div className={styles.emailinput}>
          <InputWithLabel
            text={"Password"}
            type={"password"}
            placeholder={"***"}
            onChangeInput={onhandlePassword}
          />
        </div>
        {/* <Link to="/user/dashboard"> */}
        <button type="submit">Login</button>
        {/* </Link> */}
        <ToastContainer />
      </form>
    </TopDown>
  );
};

