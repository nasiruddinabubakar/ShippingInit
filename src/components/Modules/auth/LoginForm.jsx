import { useState } from "react";
import styles from "./SignUp.module.css";
import { InputWithLabel } from "../../Molecules/InputWithLabel";

export function LoginForm() {
  const [mail, setMail] = useState("");
  function onhandleEmail(e) {
    console.log(e.target.value);
  }
  function onhandlePassword(e) {
    console.log(e.target.value);
  }
  return (
    <form className={styles.login}>
      <div className={styles.emailinput}>
        <InputWithLabel
          text={"Email Address"}
          type={"email"}
          placeholder={"jack99@gmail.com"}
          onChangeInput={onhandleEmail}
        />
      </div>
      <div className={styles.emailinput}>
        <InputWithLabel
          text={"Password"}
          type={"password"}
          placeholder={"***"}
          onChangeInput={onhandlePassword}
        />
      </div>
      <button>Login</button>
    </form>
  );
}
