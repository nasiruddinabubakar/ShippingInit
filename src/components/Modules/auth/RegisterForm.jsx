import TopDown from "../../framer/TopDown";
import { InputWithLabel, PhoneWithLabel } from "../../Molecules/InputWithLabel";
import styles from "./SignUp.module.css";
import { ToastContainer } from "react-toastify";


export default ({
  handleRegister, onHandleName, onHandleMail, onsetTemp, onHandlePassword,onHandlePhone,onHandleAddress
}) => {
  return (
    <TopDown>
      <form className={styles.login} onSubmit={handleRegister}>
        <div className={styles.emailinput}>
          <InputWithLabel
            text={"Enter Your Name"}
            type={"text"}
            placeholder={"Jack Reacher"}
            onChangeInput={onHandleName} />
        </div>
        <div className={styles.emailinput}>
          <InputWithLabel
            text={"Email Address"}
            type={"email"}
            placeholder={"jack99@gmail.com"}
            onChangeInput={onHandleMail} />
        </div>
        <div className={styles.emailinput}>
          <InputWithLabel
            text={"Password"}
            type={"password"}
            placeholder={"*****"}
            onChangeInput={onsetTemp} />
        </div>
        <div className={styles.emailinput}>
          <InputWithLabel
            text={"Confirm Password"}
            type={"password"}
            placeholder={"*****"}
            onChangeInput={onHandlePassword} />
          <ToastContainer />
        </div>
        <div className={styles.emailinput}> 
          <PhoneWithLabel 
          text={"Enter Phone Number"}
          type={"tel"}
          pattern={"\+92-[0-9]{3}-[0-9]{7}"}
          placeholder={"+92-XXX-XXXXXXX"}
          onChangeInput={onHandlePhone}/>
        </div>
        <div className={styles.emailinput}> 
          <PhoneWithLabel 
          text={"Enter Your Address"}
          type={"text"}
          placeholder={"27/2 street 1"}
          onChangeInput={onHandleAddress}/>
        </div>
        <button>Register</button>
      </form>
    </TopDown>
  );
};
