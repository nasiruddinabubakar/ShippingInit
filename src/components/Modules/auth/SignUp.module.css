.login {

    background-color: var(--color-dark--2);
    border-radius: 7px;
    padding: 2rem 3rem;
    width: 100%;
    display: flex;
    flex-direction: column;
    gap: 2rem;
    width: 48rem;
    margin: 5rem auto 0rem auto;
    transition: 0.32s ease;
  }
  header {
    height: 80px;
    width: 100%;
  }
  a{
    text-decoration: none;
  }
  .emailinput {
      color: #ececec;
    display: flex;
    flex-direction: column;
    gap: 1rem;
  }
  .emailinput input{
          width: 100%;
      padding: 0.8rem 1.2rem;
      font-family: inherit;
      font-size: 1.6rem;
      border: none;
      border-radius: 5px;
      background-color: var(--color-light--3);
      transition: all 0.2s;
  }
  .emailinput label {
   
    font-weight: 600;
    font-size: 1.6rem;
  }
  .passwordinput{
      color: #ececec;
      display: flex;
      flex-direction: column;
      gap: 0.5rem;
  }
  .passwordinput input{
      width: 100%;
  padding: 0.8rem 1.2rem;
  font-family: inherit;
  font-size: 1.6rem;
  border: none;
  border-radius: 5px;
  background-color: var(--color-light--3);
  transition: all 0.2s;
  }
  .passwordinput label {
  
  font-weight: 600;
  font-size: 1.6rem;
  }
  .login button{
      cursor: pointer;
      font-size: 1.6rem;
      font-weight: 700;
      font-family: inherit;
      width: 30%;
      border: 1px solid var(--color-brand--2);
      background-color: #00c46a;
      padding: 1rem 2rem;
      border-radius: 5px;
  }
  .login button:hover{
      font-size: 1.6rem;
      font-weight: 700;
      font-family: inherit;
      width: 30%;
      border: 1px solid var(--color-brand--2);
      background-color: #00c46a;
      padding: 1rem 2rem;
      border-radius: 5px;
  }
  p , a{
      font-size: 1.3rem;
     
      display: flex;
      color: #d6dee0;
     
     
  }
   section{
      display: flex;
      gap: 1rem;
      width: 20rem;
      margin: 1rem auto 1rem auto;
  }
  @media screen and (max-width:1366px) {
    .emailinput input{
        width: 100%;
    padding: 0.8rem 1.2rem;
    font-family: inherit;
    font-size: 1.3rem;
    border: none;
    border-radius: 5px;
    background-color: var(--color-light--3);
    transition: all 0.2s;
}
.emailinput label {
 
  font-weight: 600;
  font-size: 1.4rem;
}

  }