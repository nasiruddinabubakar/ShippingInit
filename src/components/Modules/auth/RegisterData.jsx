import { useReducer, useState } from "react";
import { toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";
import RegisterForm from "./RegisterForm";

const registerReducer = (state, { type, payload }) => {
  switch (type) {
    case "SET_NAME":
      return { ...state, name: payload };
    case "SET_MAIL":
      return { ...state, mail: payload };

    case "SET_PASSWORD":
      return { ...state, password: payload };

    default:
      return state;
  }
};

export const RegisterData = () => {
  const [registerUser, dispatch] = useReducer(registerReducer, {
    name: "",
    mail: "",
    password: "",
  });
  const [temp, setTemp] = useState("");

  function onsetTemp(e){
    setTemp(e.target.value);
  }
  
  function onHandleName(e) {
    dispatch({ type: "SET_NAME", payload: e.target.value });
  }
  function onHandleMail(e) {
    dispatch({ type: "SET_MAIL", payload: e.target.value });
  }
  function onHandlePassword(e) {
    if (temp === e.target.value)
      dispatch({ type: "SET_PASSWORD", payload: e.target.value });
  }
  const validateEmail = (email) => {
    // Regular expression for email validation
    const regex = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/;

    return regex.test(email);
  };

  async function handleRegister (e)  {
    e.preventDefault();
    const notify = (text) => {
      toast.error(text, {
        position: toast.POSITION.TOP_RIGHT,
        className: "toast_message",
      });
    };
    console.log(validateEmail(registerUser.mail));
    if (validateEmail(registerUser.mail) === false) {
      notify("Email is not valid");
      return;
    } else {
    } 
    if (registerUser.password === "") {
      notify("Password Doesnt match");
      return;
    }
    
    const res = await postData('http://127.0.0.1:5000/api/users/register',registerUser);
    console.log(res);  
    if(res){
       
        toast.success("Account Created",{
          position: toast.POSITION.TOP_RIGHT,
          className: "toast_message",
        })
      }
    
  };
  return (
    <RegisterForm
      handleRegister={handleRegister}
      onHandleName={onHandleName}
      onHandleMail={onHandleMail}
      onsetTemp={onsetTemp}
      onHandlePassword={onHandlePassword}
    />
  );
};

async function postData(url = "", data = {}) {
  const response = await fetch(url, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify(data),
  })
return await response.json();
 
}
