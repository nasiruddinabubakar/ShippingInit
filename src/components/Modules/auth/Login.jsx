import { useState } from "react";
import { RegisterData } from "./RegisterData";
import { LoginForm } from "./LoginForm";

import { Header } from "../../UI/Header";
export const Login = () => {
  const [isLogin, setIsLogin] = useState(true);
  const LoginRegister = isLogin ? LoginForm : RegisterData;

  function onHandleLogin() {
    setIsLogin((isLogin) => !isLogin);
  }
  return (
    <>
    
      <div className="Main">
     <Header/>
      {<LoginRegister />}
      <ToggleLoginSignup onHandleLogin={onHandleLogin} isLogin={isLogin} />
      </div>
    </>
  );
};
export function ToggleLoginSignup({ onHandleLogin, isLogin }) {
  return (
    <section>
      {isLogin ? (
        <>
          <p>New Account?</p>
          <a href="#" onClick={onHandleLogin}>
            Register Now
          </a>
        </>
      ) : (
        <>
          {" "}
          <p>Already a User?</p>
          <a href="#" onClick={onHandleLogin}>
            Login Now
          </a>
        </>
      )}
    </section>
  );
}

