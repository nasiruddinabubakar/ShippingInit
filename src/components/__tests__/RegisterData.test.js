import { render, screen, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer';
import { LoginData } from '../Modules/auth/LoginData';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../app/store';
import RegisterForm from '../Modules/auth/RegisterForm';
import { log } from 'console';




afterEach(()=>{
  cleanup();
})




test('should render LoginData component', () => {
  render(
    <BrowserRouter>
    <Provider store={store}>
      <LoginData />
      </Provider>
    </BrowserRouter>
  );
  const loginElement = screen.getByTestId('login');
  expect(loginElement).toBeInTheDocument();
  expect(loginElement).toHaveTextContent('Email Address');
  expect(loginElement).toHaveTextContent('Password');
  expect(loginElement).toContainHTML('input')
});

test('it should render RegisterData component', () => {
  render(
    <BrowserRouter>
    <Provider store={store}>
      <RegisterForm />
      </Provider>
    </BrowserRouter>
  );
  const registerElement = screen.getByTestId('register');
  expect(registerElement).toBeInTheDocument();
});

test('matches snapshot', () => {
  const tree = renderer.create(
    <BrowserRouter>
    <Provider store={store}>
      <LoginData />
      </Provider>
    </BrowserRouter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});