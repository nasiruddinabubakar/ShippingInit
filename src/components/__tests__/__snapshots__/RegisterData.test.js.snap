// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  style={
    Object {
      "marginTop": "0rem",
    }
  }
>
  <form
    className="login"
    data-testid="login"
    onSubmit={[Function]}
  >
    <div
      className="emailinput"
    >
      <label>
        Email Address
      </label>
      <input
        onChange={[Function]}
        placeholder="jack99@gmail.com"
        type="email"
      />
    </div>
    <div
      className="emailinput"
    >
      <label>
        Password
      </label>
      <input
        onChange={[Function]}
        placeholder="***"
        type="password"
      />
    </div>
    <button
      type="submit"
    >
      Login
    </button>
    <div
      className="Toastify"
    />
  </form>
</div>
`;
